### PR TITLE
Add find unassigned menu component

### LINKED ISSUE #122: Find unassigned
We want to make it possible for users to find unassigned geographic areas. This will work similarly to Ctrl-F in a web page.

AC:
1. There is a find button in top-right. Clicking pops down a dialog.
2. The dialog displays a count of unassigned geographies (e.g. `20 found`)
3. Hitting an arrow key zooms to the extent of an unassigned geography and changes the dialog display (e.g. `x/20`. Hitting → would take you to 1/20 and ← would take you to 20/20.)
4. Hitting the arrow keys cycles through the other shapes and changes the numbers accordingly (e.g. 2/20, 3/20, etc.) 
- [wireframe series](https://app.abstract.com/projects/9f988240-1dad-11e8-9a40-cba930331af7/branches/master/commits/238ea75510f11bde6dcba5f841bd99aa326bd167/files/DCA38482-FC7F-4F48-BECD-44E07D9F7443/layers/62C5E14E-176D-4DD3-B038-4810D7B0EC7D?mode=design)

### LINKED ISSUE #462: Redo focus is oval-shaped
The focus on the redo button is oval shaped whereas the focus on the undo button is a circle. The redo button should be a circle as well.

Undo:

Redo:

### PR DESCRIPTION
**Pull request by maurizi**
## Overview

Adds a new button to the project header to toggle a find unassigned menu, which allows zooming the map between unassigned areas

### Checklist

- [x] Description of PR is in an appropriate section of `CHANGELOG.md` and grouped with similar changes, if possible

### Demo

localhost_3003_projects_e1cb7dfc-eafb-4566-bbec-9c325aae58ce
localhost_3003_projects_e1cb7dfc-eafb-4566-bbec-9c325aae58ce (1)
localhost_3003_projects_e1cb7dfc-eafb-4566-bbec-9c325aae58ce (2)

## Testing Instructions

- `scripts/update`
- Go to a project
- Click the find button

Closes #122 
Closes #462

### CODE REVIEW

**Review by kshepard — CHANGES_REQUESTED** [review]
I'm able to generate the following crash in a couple different ways:

unassigned-error

The easiest way to reproduce is to click the forward/next buttons when there are no unassigned polygons found. Another way is harder, so I added a GIF to demonstrate:

unassigned-error

Seems to happen when there are multiple unassigned polygons, and you are on the final one and go ahead and assign it.

I also saw some weirdness at one point, that I wasn't able to reproduce consistently, where it was telling me there was an unassigned polygon, but it would zoom to an area that was fully assigned. Sorry that I'm unable to provide instructions for reproducing it, but just something to keep an eye out for as this gets exercised some more during testing.

**Comment by tgilcrest** [comment]
Does this work for unassigned blocks as well?

File: src/client/reducers/districtDrawing.ts [hunk 1]
```diff
@@ -20,13 +20,16 @@ import {
...
+  replaceSelectedGeounits,
+  toggleFind,
+  setFindIndex
 } from "../actions/districtDrawing";
 import { SelectionTool } from "../actions/districtDrawing";
 import { resetProjectState } from "../actions/root";
 import { GeoUnits, GeoUnitsForLevel, LockedDistricts } from "../../shared/entities";
 import { ProjectState, initialProjectState } from "./project";
 import { setFeaturesSelectedFromGeoUnits } from "../components/map";
+import { isSwitchStatement } from "typescript";
```
Comment (kshepard) [inline src/client/reducers/districtDrawing.ts:32]:
```suggestion
```

File: src/client/components/map/FindMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import { polygon } from "@turf/helpers";
+
+import Icon from "../Icon";
+
+import { State } from "../../reducers";
+import { DistrictsGeoJSON } from "../../types";
+import { setFindIndex, toggleFind } from "../../actions/districtDrawing";
+import store from "../../store";
+import { destructureResource } from "../../functions";
+import { useEffect, useState } from "react";
```
Comment (kshepard) [inline src/client/components/map/FindMenu.tsx:15]:
```suggestion
import { useEffect } from "react";
```

**Review by kshepard — APPROVED** [review]
Looks good. I just added a couple minor suggestions to remove unused imports. I again was able to come up with a single scenario where it was telling me that there was an unassigned polygon and zoomed me to an area where nothing was unassigned. Then I refreshed the page and it was gone. I tried to reproduce it a dozen different ways and was unable. Not sure what's up with that, but it seems pretty hard to come across, and if it is indeed an issue, perhaps we can rout it out better when this gets to staging.

**Comment by kshepard** [comment]
> Does this work for unassigned blocks as well?

Yep, it's geolevel-agostic, and just works with the district GeoJSON itself.

**Pull request merged**